rpc/src/v1/impls/eth_filter.rs

```diff
@@ -52,6 +52,9 @@ pub trait Filterable {
 
 	/// Get a reference to the poll manager.
 	fn polls(&self) -> &Mutex<PollManager<PollFilter>>;
+
+	/// Get removed logs within route from the given block to the nearest canon block, not including the canon block. Also returns how many logs have been traversed.
+	fn canon_logs(&self, block_hash: H256, filter: &EthcoreFilter) -> (Vec<Log>, u64);
```

Comment:
Can we rename this to something like `removed_logs` or `reorged_logs`? Something other than `canon_logs` since we're returning logs that are not canon.